pkg/component/kubernetes/apiserverexposure/sni.go

```diff
@@ -40,6 +40,8 @@ const (
 	// MutualTLSServiceNameSuffix is used to create a second service instance for
 	// use with mutual tls authentication from istio using the ca-front-proxy secrets.
 	MutualTLSServiceNameSuffix = "-mtls"
+	// UpgradeServiceNameSuffix is used to create service instance for connections which set the Upgrade HTTP header.
+	UpgradeServiceNameSuffix = "-upgrade"
```

Comment:
Nit: The suffix leads to an interesting name `kube-apiserver-upgrade`, which might cause confusion. Therefore, I would prefer something like `-spdy`, `-speedy` or `-connection-upgrade`, but I have not found the perfect name, yet. Maybe you are more creative.
(I associate `upgrade` with a rolling blue/green update and not with SPDY.)

Reply:
Naming is hard 😅 
I chose `-connection-upgrade` because the endpoint is not only used for SPDY but for upgrades to websocket too.